### PR TITLE
core: rpmb: fix initialization of new rpmb data

### PR DESCRIPTION
**Pull request by etienne-lms**
i tried to optimize `memset()` use rather than a big `memset(blk_buf, 0, TMP_BUF_SIZE);` in the loop.

### CODE REVIEW

**Comment by jforissier** [comment]
`Tested-by: Jerome Forissier <jerome@forissier.org> (QEMU)`

**Comment by etienne-lms** [comment]
Tested-by tag applied.

File: core/tee/tee_rpmb_fs.c [hunk 1]
```diff
@@ -2410,20 +2414,26 @@ static TEE_Result update_write_helper(struct rpmb_file_handle *fh,
...
 				copy_size -= offset;
 			}
 			copy_size = MIN(copy_size, rem_size);
 
-			memcpy(dst, rem_buf, copy_size);
+			memcpy(copy_dst, rem_buf, copy_size);
 			rem_buf += copy_size;
 			rem_size -= copy_size;
+
+			memset(blk_buf + rd_size, 0, offset - rd_size);
```
Comment (jenswi-linaro) [inline core/tee/tee_rpmb_fs.c:2434]: Please add a comment explaining the `memset()` strategy here. It is quite hard to figure out what's going on here.

Comment (etienne-lms) [reply core/tee/tee_rpmb_fs.c:2434]: ok

**Comment by etienne-lms** [comment]
added an inline comments

**Comment by jenswi-linaro** [comment]
`Reviewed-by: Jens Wiklander <jens.wiklander@linaro.org>`

**Comment by etienne-lms** [comment]
tag applied.

**Comment by jforissier** [comment]
Thanks for the fix.

**Comment by etienne-lms** [comment]
Sorry for the bug.

**Comment by jenswi-linaro** [comment]
No worries.

**Pull request merged**